File: medical-ai-app/backend/app/schemas/report(1).py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class ReportOptions(BaseModel):
    template_type: str = "standard"
    include_images: bool = True
    include_metrics: bool = True
    include_notes: bool = True
    mask_personal_info: bool = False
    show_logo: bool = True


class ReportGenerateRequest(ReportOptions):
    pass


class ReportRead(BaseModel):
    id: int
    case_id: int
    template_type: str
    include_images: bool
    include_metrics: bool
    include_notes: bool
    mask_personal_info: bool
    pdf_path: str
    created_at: datetime

    class Config:
        from_attributes = True


class ReportPreviewResponse(BaseModel):
    report: ReportRead
    case: dict[str, Any]
    prediction: dict[str, Any] | None = None

File: medical-ai-app/frontend/src/types/report(1).ts
export interface ReportRecord {
  id: number;
  case_id: number;
  template_type: string;
  include_images: boolean;
  include_metrics: boolean;
  include_notes: boolean;
  mask_personal_info: boolean;
  pdf_path: string;
  created_at: string;
}

export interface ReportResponse {
  report: ReportRecord;
  case: {
    id: number;
    case_code: string;
    patient_name: string;
    patient_identifier: string;
    study_date: string;
    module_type: string;
    status: string;
    notes: string;
    input_file_path: string;
  };
  prediction?: {
    predicted_label: string;
    confidence: number;
    summary: string;
    metrics: Record<string, unknown>;
    artifacts: Record<string, string>;
    created_at: string;
  } | null;
}
